### PR TITLE
Project Creation Vue

### PR DESCRIPTION
**Pull request by felixoi**

### CODE REVIEW

File: oreClient/src/main/assets/stores/top.js [hunk 1]
```diff
@@ -13,6 +13,11 @@ const mutations = {
             message: payload.message
         })
     },
+    addAlerts(state, payload) {
+        state.alerts[payload.level] = state.alerts[payload.level].concat(payload.messages.map(m => {
```
Comment (Katrix) [inline oreClient/src/main/assets/stores/top.js:19]: Is this safe in terms of reactivity and Vue?

File: oreClient/src/main/assets/pages/project/NewProject.vue [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                        </div>
+
+                        <div class="form-group">
+                            <label for="projectPluginId">Plugin id</label>
+                            <input type="text" id="projectPluginId" name="pluginId" class="form-control" v-model.trim="pluginId" required>
+                        </div>
+
+                        <div class="form-group">
+                            <label for="projectCategory">Project category</label>
+                            <select class="form-control" v-model="category" required>
```
Comment (Katrix) [inline oreClient/src/main/assets/pages/project/NewProject.vue]: Missing id

File: oreClient/src/main/assets/api.js [hunk 3]
```diff
@@ -27,6 +28,11 @@ export class API {
                         withCredentials: true
                     },
                 }).done((data) => {
+                    //store.commit({
```
Comment (Katrix) [inline oreClient/src/main/assets/api.js]: Did you ever get this working?

File: oreClient/src/main/assets/pages/project/NewProject.vue [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                    });
+                } else {
+                    API.request("projects", "POST", {
+                        name: this.projectName,
+                        plugin_id: this.pluginId,
+                        category: this.category,
+                        description: this.projectDescription,
+                        owner_name: this.owner
+                    }).then((data) => {
+                        this.$router.push({path: `/${data.namespace.owner}/${data.namespace.slug}`})
```
Comment (Katrix) [inline oreClient/src/main/assets/pages/project/NewProject.vue]: Why not do name based routing, and also pass it extra info we have about the project. We have the whole fetched project, so might as well just pass it in. Lookup by namespace is also a bit slower currently.

File: oreClient/src/main/assets/api.js (hunk 3 above)
Comment (felixoi) [reply oreClient/src/main/assets/api.js]: yeah that works, remove the comments and it should work instantly

**Pull request merged**